docs/using-est-with-certd.md

```diff
@@ -0,0 +1,187 @@
+
+# Authentication with username and password.
+username = "username"
+password = "password"
+
+# Sample configuration of a single EST-issued certificate.
+# Replace `name` with the desired certificate name.
+[cert_issuance.name]
+
+# Identifies this certificate as being issued over EST.
+method = "est"
+
+# Optional certificate common name. Defaults to the CSR's common name if not provided.
+common_name = "common name"
+
+# Optional number of days between certificate issuance and expiry. Defaults to 30 if not provided.
+expiry_days = 30
+
+# Optional EST URL to issue this certificate. Defaults to the `default` URL in `[cert_issuance.est.urls]`
+# if not provided. The URL must be provided either here or in default, i.e. certd will fail if no URL is
+# provided here and no default exists.
+url = "https://est.example.com/.well-known/est"
+
+# It is also possible to configure auth separately for each certificate. The options are the
+# same as in the global EST configuration and override the global configuration for their corresponding
+# certificate.
+identity_cert = "identity-cert-id"
+identity_pk = "identity-cert-pk-id"
+
+bootstrap_identity_cert = "bootstrap-identity-cert-id"
+bootstrap_identity_pk = "bootstrap-identity-pk-id"
+
+username = "username"
+password = "password"
+```
+
+## Sample Configuration
+
+Below are sample configuration files that can be used as starting points to test EST certificate issuance.
+
+The certd configuration file configures EST certificate issuance.
+
+`/etc/aziot/certd/config.d/test-est-cert.toml`
+
+```toml
+# Configure the trusted root CA certificate in the global EST options. This section is optional
+# if the EST server's TLS certificate is already trusted by the system's CA certificates.
+[cert_issuance.est]
+trusted_certs = ["est-server-ca"]
+
+[cert_issuance.est.urls]
+
+[preloaded_certs]
+est-server-ca = "file:///path/to/file.pem"
+
+# Configure the issuance of the EST test certificate.
+[cert_issuance.test-est-cert]
+method = "est"
+
+# Example with the EST path segment `test-est-cert` in the URL.
+# The path will depend on server configuration.
+url = "https://est.example.com/.well-known/est/test-est-cert"
+
+# This example will use the EST bootstrap certificate and key to authenticate,
+# but a cert and key ID for the identity certificate must still be provided.
+# These credentials will automatically be created, so it is sufficient to provide
+# a set of unique IDs that will not be used for any other credentials.
+identity_cert = "test-est-cert-identity-id"
+identity_pk = "test-est-cert-identity-id"
+
+# The credentials to use upon initial authentication with the EST server.
+# After that, certd will automatically create, use, and renew identity_cert and identity_pk.
+bootstrap_identity_cert = "test-est-bootstrap-cert-id"
+bootstrap_identity_pk = "test-est-bootstrap-key-id"
+
+# Load the bootstrap certificate from a file into certd.
+# This file must be readable by aziotcs.
+[preloaded_certs]
```

Comment:
Just to confirm - this preloaded_certs seems to be in multiple places. I suppose it is a top level item, so you have just put it in the logical places?

Reply:
Yes, it's fine to put it in multiple places.